lib/test_queue/runner.rb

```diff
@@ -427,40 +427,53 @@ def distribute_queue
         else
           sock = @server.accept
           cmd = sock.gets.strip
+
+          cmd =~ /^TOKEN=(\w+)/
```

Comment:
I agree with @bhuga that stripping off the token here would be clearer. Maybe it'd also be good to use `Regexp#match` or similar instead of dealing with the magic `$1` variable.